src/CameraEnvironmental.h

```diff
@@ -3,5 +3,46 @@
 #include "ICamera.h"
 #include "ray.h"
 
-// --- IMPLENET class CCameraEnvironmental ---
+#define _USE_MATH_DEFINES
+#include <math.h>
+// --- IMPLEMENT class CCameraEnvironmental ---
 // --- PUT YOUR CODE HERE ---
+
+class CCameraEnvironmental : public ICamera
+{
+public:
+	CCameraEnvironmental(Size resolution, const Vec3f& pos, float angle)
+		: ICamera(resolution)
+		, m_pos(pos)
+	{
+	}
+	virtual ~CCameraEnvironmental(void) = default;
+
+	virtual void InitRay(Ray& ray, int x, int y) override
```

Comment:
I like your implementation, it general it looks correct. After a closer look, there are some error. You need to spend a bit more time for debugging.